torchdata/datapipes/iter/util/prefetcher.py

```diff
@@ -102,6 +102,3 @@ def reset(self):
         if self.thread is not None:
             self.prefetch_data.run_prefetcher = False
             self.thread.join()
-
-    def reset_iterator(self):
-        self.reset()
```

Comment:
Yeah, I found the same issue as `reset` will be called automatically at the beginning of `__iter__`.